driver-core/src/main/com/mongodb/internal/connection/InternalConnection.java

```diff
@@ -65,6 +65,15 @@
      */
     boolean isClosed();
 
+    /**
+     * Gets the generation of this connection.  This can be used by connection pools to track whether the connection is stale.
+     *
+     * @return the generation.
+     */
+    default int getGeneration() {
```

Comment:
Had to add this new method to the interface, but gave it a default implementation that indicates it's unsupported.  This will be the case for connections that don't come from a pool (which currently is only the monitoring connection)